### PR TITLE
Memleak queue fixes

### PR DESCRIPTION
**Pull request by akallabeth**
* Fix a memory leak in gfx on error
* Now clearing message queue on free just as Hash and ArrayList
* dynamic channel message queue now frees objects still in the queue
* Clear the client channel queue and free data

### CODE REVIEW

File: winpr/libwinpr/utils/collections/MessageQueue.c [hunk 1]
```diff
@@ -227,6 +227,8 @@ void MessageQueue_Free(wMessageQueue* queue)
 	if (!queue)
 		return;
 
+	MessageQueue_Clear(queue);
+
```
Comment (hardening) [inline winpr/libwinpr/utils/collections/MessageQueue.c]: good catch

**Comment by hardening** [comment]
@akallabeth just a matter of taste but I prefer
```
status = op()
if (status)
  .....
```
over
```
status = op()

if (status)
  .....
```

IMHO it looks better when the operation and the test are grouped in the same block.

**Comment by akallabeth** [comment]
@mfleisz Already fixed.

**Comment by mfleisz** [comment]
@freerdp-bot test

**Pull request merged**